### DISCUSSION TITLE
Fantastic software

### DISCUSSION THREAD

**Discussion by JimBol22** [Hofknecht/SystemTrayMenu#480] [General]
Very happy I found this software! Using a Mac and Windows, I was surprised that Windows doesn't offer spring-loaded folders or a column view for the folders in file explorer, which makes stepping through your folders much faster and easier. Also, I loved using Xmenu on MacOS which integrates this capability into the menu bar (it is default in the dock). Anyway, through a search I found SystemTrayMenu and I am super-pleased with it. It is fast, works as expected and I love the ability to open it from the cursor position (for which I use the  middle button of the Logitech mouse (assigned the Ctrl+Windows keys to that button). I have added any and all folders from where I want to start the search to a single folder which is the one SystemTrayMenu opens upon activating the app. From there I can then swiftly navigate to wherever I like. Shortcuts to folders are supported which makes this very easy and non-disruptive to setup (no need to move actual folders)! 
So thank you, and I wanted to let you and new users know how pleased I am with your software; I hope others will find this feedback useful and are encouraged to install SystemTrayMenu.

**Comment by JimBol22** [comment]
@Hofknecht - Do you have any plans to develop this for macOS also? Would be great!

**Comment by Hofknecht** [comment]
@JimBol22 
Thank you for the question!

@topeterk is currently working at
https://github.com/Hofknecht/SystemTrayMenu/tree/v2.x

with which we are theoretically going in the direction of supporting it on mac os or linux in the future. But that is still a long way off

**Reply by JimBol22** [reply]
Dear Markus or Peter, 

Is there an update on v2? I have found a good alternative solution on the Mac (Folder Peek) but STM is still my favourite 🥇 . 

Thank you.

**Reply by topeterk** [reply]
Thank you, thats always nice to hear when people like the software and give us any kind of feedback.

Markus' mentioned v2 version is a **remaster of v1** using WPF: https://github.com/Hofknecht/SystemTrayMenu/tree/v2.x
Status is tracked here: https://github.com/Hofknecht/SystemTrayMenu/projects/19
On my fork, there is another v2 version which is a **remake of the above** using Avalonia: https://github.com/topeterk/SystemTrayMenu

The name or final version is not final or how they will be published in the future, so we run both under the name "v2" at the moment.
Since WPF version is still heavily relying on Windows, this will still not run Linux or MacOS.
However, the Avalonia version is published for each platform (Windows, Linux and MacOS).

Since I do not own a Mac and Apple doesn't provide free-of-charge test machines, I cannot test anything for MacOS.
For Linux I tested this on Ubuntu 22.04 but it *should* support (to some extend) GTK, KDE and LxQt.

Today, I created the first pipeline to build first portable deliveries for testing.
[SystemTrayMenu.win-x64-v2-PreRelease.zip](https://github.com/user-attachments/files/15745710/SystemTrayMenu.win-x64-v2-PreRelease.zip)
[SystemTrayMenu.linux-x64-v2-PreRelease.tar.gz](https://github.com/user-attachments/files/15745712/SystemTrayMenu.linux-x64-v2-PreRelease.tar.gz)
[SystemTrayMenu.osx-x64-v2-PreRelease.tar.gz](https://github.com/user-attachments/files/15745713/SystemTrayMenu.osx-x64-v2-PreRelease.tar.gz)

> v2 Build 06/08/24 Avalonia Fork
> commit https://github.com/topeterk/SystemTrayMenu/commit/87ee2422abca24716e9b27bec06992ac60209cd7

**Please note: .NET 8 Runtime must be installed on the System**

**Reply by JimBol22** [reply]
Hi Peter,

This is promising, and I’d be happy to test it on Mac. However, I am not proficient with Developer tools, so I need an actual release for Mac ARM64 (I only see SystemTrayMenu 1.3.5.0 on all project pages / links). I tried a build from the csproj file on my system, but did not succeed. So the release (app) or clear instructions on how to build it myself would be great.

**Reply by topeterk** [reply]
Hi,
at first I started with x64 builds only but I added arm64 build as well:

[SystemTrayMenu.osx-arm64-v2-PreRelease.tar.gz](https://github.com/user-attachments/files/15751305/SystemTrayMenu.osx-arm64-v2-PreRelease.tar.gz)

> v2 Build 06/09/24 Avalonia Fork
> commit https://github.com/topeterk/SystemTrayMenu/commit/d653c673d78d55c2fb934f4de486ea463f460010

This shouls work on your ARM64. So when .NET 8 is installed it should be enough to extract the archive somewhere on your system and I it looks like the file "SystemTrayMenu" should be able to run the application.

There is no official release yet, that is why I posted the development builds here.
The builds are taken from Azure Pipeline https://dev.azure.com/topeterk/SystemTrayMenu/_build?definitionId=2
> In case someone wants to download specific or newer versions:
After clicking on a build file should be accessible (when build succeeded) after clicking on:
![image](https://github.com/Hofknecht/SystemTrayMenu/assets/4932525/f6f36fae-d65a-4aef-9d5b-26044cbc704f)
Then at the button "..." on the right the files can be downloaded.

**Comment by JimBol22** [comment]
Thank you and I had a good look at it. Following my feedback on the Mac version:

1. **Running via Terminal**: 
    - I could not find a standalone app, so I am running STM through Terminal. Note that closing Terminal also closes STM.

2. **Building from CSPROJ File**:
    - I managed to create a version from the `CSPROJ` file in `SystemTrayMenu-2.x_Avalonia` and also used the one in `SystemTrayMenu.osx-arm64-v2-PreRelease`. The results were the same for both and are described below.
    - BTW, I tried to find the app version that runs without Terminal using the link you provided. However, after clicking on the build file (assuming it succeeded), there was no "..." button on the right for downloading files.

3. **Directory Hover Issue**:
    - In STM on the Mac, when I first hover over the directories, nothing happens. I need to click on a directory before it will open and show its contents.

4. **Disk Content Issue**:
    - STM shows only the contents of an external disk connected to the Mac. The local drive and NAS do not open in STM. Clicking on these disks and directories will open them in Finder, but after opening the directory in Finder, STM still does not show contents when hovering over it.

**Reply by topeterk** [reply]
Hi, interesting..
1. I would have expected that the file `SystemTrayMenu` can be executed. Have you tried to start this one?
1. I think you need to hover with the mouse over the line. I also think this is bad design as finding these files are pretty weird and also looks different when you access the website as anonymous or as logged in user:  
![image](https://github.com/Hofknecht/SystemTrayMenu/assets/4932525/992c91ab-d26b-4ee1-8291-ebef44d0859a)
1. Is this always or just the very first time? The focus and active window handling is not completely implemented yet, so not sure if this is actually a bug or if it is just not finished/polished yet. But good to know anyway that this is happending.
1. For Windows there are several special code sections for network drives but on Linux (and I think on Mac as well?) there is only one big file system with everything mounted there. So the applications don't have to care or know what kind of drive it is. This sounds a bit strange that this does not work. Or does MacOS have such a thing like "drives" as Windows or manages somehow multiple file systems? 
 Double click should open files and folder but maybe the setting is wrong that allows opening it with a single click, then sub menus should only open by hovering or by selecting the entry via keyboard (which maybe interferes with the focus and activation window handling mentioned at number 3) .

**Reply by JeckPeppinger** [reply]
hi, is there an update on the Linux work? Can this be tested on ubuntu 24.10? Thx!

**Reply by topeterk** [reply]
Sorry, but there were no further updates yet. Also no tests were done with Ubunut 24.10 release yet.

**Reply by Xalalau** [reply]
I'm looking forward the Linux version. So far there's no replacement for your software, it's unique.

**Comment by BugFixNo1** [comment]
incredible cool thing! thank you for the effort spent!
